lib/asyncapiSchemaFormatParser.js

```diff
@@ -0,0 +1,69 @@
+const Ajv = require('ajv');
+const ParserError = require('./errors/parser-error');
+const asyncapi = require('@asyncapi/specs');
+const { improveAjvErrors } = require('./utils');
+
+module.exports = {
+  parse,
+  getMimeTypes
+};
+
+async function parse({ message, originalAsyncAPIDocument, fileFormat, parsedAsyncAPIDocument, pathToPayload }) {
+  const ajv = new Ajv({
+    jsonPointers: true,
+    allErrors: true,
+    schemaId: 'id',
+    logger: false,
+  });
+  const payloadSchema = preparePayloadSchema(asyncapi[parsedAsyncAPIDocument.asyncapi]);
+  const validate = ajv.compile(payloadSchema);
+  const valid = validate(message.payload);
+
+  if (!valid) throw new ParserError({
+    type: 'schema-validation-errors',
+    title: 'This is not a valid AsyncAPI Schema Object.',
+    parsedJSON: parsedAsyncAPIDocument,
+    validationErrors: improveAjvErrors(addFullPathToDataPath(validate.errors, pathToPayload), originalAsyncAPIDocument, fileFormat),
+  });
+}
+
+function getMimeTypes() {
+  return [
+    'application/vnd.aai.asyncapi;version=2.0.0',
+    'application/vnd.aai.asyncapi+json;version=2.0.0',
+    'application/vnd.aai.asyncapi+yaml;version=2.0.0',
+    'application/schema;version=draft-07',
+    'application/schema+json;version=draft-07',
+    'application/schema+yaml;version=draft-07',
+  ];
+}
+
+/**
+ * To validate schema of the payload we just need a small portion of official AsyncAPI spec JSON Schema, the definition of the schema must be
+ * a main part of the JSON Schema
+ * 
+ * @param  {Object} asyncapiSchema AsyncAPI specification JSON Schema
+ * @returns {Object} valid JSON Schema document describing format of AsyncAPI-valid schema for message payload
+ */
+function preparePayloadSchema(asyncapiSchema) {
+  return {
+    $ref: '#/definitions/schema',
```

Comment:
That's smart 👏

Reply:
![](https://media.giphy.com/media/3o7aD4ubUVr8EkgQF2/giphy-downsized.gif)